### ISSUE TITLE
_THEOS_ESCAPED_STAGING_DIR formatting issue on Linux

### ISSUE THREAD

**Issue by candoizo** [theos/theos#525]
(Thanks for reporting an issue! Please make sure you click the link above to view the issue guidelines, then fill out the blanks below.)

Thanks a ton for all the major improvements and fixes you've been putting into this project!

What are the steps to reproduce this issue?
-------------------------------------------
1. Install theos on Linux
2. Try to build a package with make
3. Error when theos tries to put the .deb together because the path it is told to check will never exist

What happens?
-------------
Package does not get built because of a misformatted path, `_THEOS_ESCAPED_STAGING_DIR` is introducing a space on Linux where they should be none causing theos to run checks in the wrong location. It's the same outcome regardless of where the control file is, `./layout/DEBIAN/` or in the project root. 

What were you expecting to happen?
----------------------------------
I expect this variable should not introduce an escaped space on the end on Linux so that theos can find what it copies into the staging directory.

Any logs, error output, etc?
----------------------------
Error message from a default nic.pl tweak running  `make clean package`
```
==> Error: make package requires you to have a layout/ directory in the project root, containing the basic package structure, or a control file in the project root describing the package.
bash: /home/user/Code/ios/test/.theos/_ /DEBIAN/control: No such file or directory.
```

Troubleshooting + Potential Solution?
-------------------

I created a blank tweak with `nic.pl` then logged some variables during a build to look for where it's was running into trouble.

```
THEOS_STAGING_DIR_NAME="_"
THEOS_STAGING_DIR="/home/user/Code/ios/test/.theos/_"

_THEOS_ESCAPED_STAGING_DIR="/home/user/Code/ios/test/.theos/_\ "
```

First two look fine, but as I mention `_THEOS_ESCAPED_STAGING_DIR` shows a quirk on the end causing this issue. 

On line 251 in common.mk is where it's first constructed, but there not much clear reason an EOL is misinterpreted as matching a space in the `subst` command, so I considered it might have to do with how this `_SPACE` variable is being constructed:
```
THEOS_STAGING_DIR ?= $(_THEOS_LOCAL_DATA_DIR)/$(THEOS_STAGING_DIR_NAME)
_SPACE :=
_SPACE +=
_THEOS_ESCAPED_STAGING_DIR = $(subst $(_SPACE),\ ,$(THEOS_STAGING_DIR))
```

For reasons unknown it seems to append two separate "blank" lines which I expect is where it's getting the idea to match probably EOL terminators. So, I tried a simple change to how `_SPACE` was declared and that worked exactly how I expected allowing my build to finish with no errors. That change is below here for your reference
```
THEOS_STAGING_DIR ?= $(_THEOS_LOCAL_DATA_DIR)/$(THEOS_STAGING_DIR_NAME)
_SPACE = " "
_THEOS_ESCAPED_STAGING_DIR = $(subst $(_SPACE),\ ,$(THEOS_STAGING_DIR))
```

If there is a reason the `_SPACE` was written this way originally its not clear to me, maybe some quirk on a different OS. Seems like it could be innocent enough to fix otherwise

What versions of software are you using?
----------------------------------------
**Operating System:** `Linux 5.7.4-arch1-1 x86_64 GNU/Linux`

**Toolchain Version:** n/a

**SDK Version:** n/a

**Comment by kabiroberai** [comment]
Are you using the latest version of Theos? The way `_SPACE` is declared was changed a while ago. Please update if you're on an older version (make sure you're on the master branch), and check whether this issue has been resolved for you.

**Comment by kirb** [comment]
Yep - this was fixed some time ago in 56ec649. GNU Make 4.3 decided to break this specific method of getting a variable consisting of just a space. Using a path with spaces doesn’t really work in Make anyway, and I don’t entirely think it’s worth all the trouble to try and force it to work, so I feel like all this escaping stuff could just be removed.

**Comment by kirb** [comment]
Closing since I’m pretty sure this was already resolved, and we removed support for paths with spaces in dd7f2f4.
